jupyterhub/singleuser.py

```diff
@@ -448,6 +448,39 @@ def _default_client(self):
 
     server_name = Unicode()
 
+    async def send_spawn_callback(self):
+        # TODO: doesn't work, not sure if auth is correct, etc
+        # TODO: reduce copy and paste here
+        # copied from notify_activity
+        client = self.hub_http_client
+
+        async def notify():
+            self.log.debug("Notifying Hub of readyness via spawn_callback")
+            req = HTTPRequest(
+                url=self.hub_activity_url,
```

Comment:
spawn_callback_url?

Reply:
And only call it if `spawn_callback_url` is set, which effectively implements single-user -> hub backward compatiblity (not the other way around)